### PR TITLE
ger implementation, 32/64 split

### PR DESCRIPTION
**Pull request by jacksonmowry**
Closes: #39 

I split out the 32/64 bit implementations but with the way Mojo currently works (at least as far as I can tell) we'll have to also split the tests (unless we want to essentially duplicate all the code within a test function for both precision levels).

I guess the compiler is unable to infer that the comptime `dtype` parameter is either `DType.float32` or `DType.float64` meaning passing a buffer of type `UnsafePointer[Dtype.float32, ImmutAnyOrigin]` is not equivalent to `UnsafePointer[dtype, ImmutAnyOrigin]` with `dtype == DType.float32`. Hopefully I'm just missing something and we can go back to having a single test.

This implementation also assumes we want to split the cpu dispatch functions into `sger` and `dger`, let me know if we'd like to still just have 1 dispatch function with a `dtype` template parameter.

### CODE REVIEW

File: test-level2.mojo [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            py_y.append(y[i])
+
+        var sp_res: PythonObject
+        # ger - float32
+        if dtype == DType.float32:
+            np_a = np.array(py_a, dtype=np.float32).reshape(m,n)
+            np_x = np.array(py_x, dtype=np.float32)
+            np_y = np.array(py_y, dtype=np.float32)
+            sp_res = sp_blas.sger(alpha, np_x, np_y, 1, 1, np_a)
+        if dtype == DType.float64:
```
Comment (tdehoff) [inline test-level2.mojo]: This should be `elif`
